cmd/root.go

```diff
@@ -0,0 +1,295 @@
+	if err != nil {
+		return nil, err
+	}
+	return rjs.Items, nil
+}
+
+func getRadixDeploymentsInNamespaces(kubeClient *kube.Kube, namespaces []string) ([]v1.RadixDeployment, error) {
+	rdsForRr := make([]v1.RadixDeployment, 0)
+	for _, ns := range namespaces {
+		rds, err := kubeClient.RadixClient().RadixV1().RadixDeployments(ns).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return nil, err
+		}
+		rdsForRr = append(rdsForRr, rds.Items...)
+	}
+	return rdsForRr, nil
+}
+
+func getRuntimeNamespaces(ra *v1.RadixApplication) []string {
+	namespaces := make([]string, 0)
+	for _, env := range ra.Spec.Environments {
+		namespaces = append(namespaces, utils.GetEnvironmentNamespace(ra.Name, env.Name))
+	}
+	return namespaces
+}
+
+func getRadixApplication(kubeClient *kube.Kube, appName string) (*v1.RadixApplication, error) {
+	return kubeClient.RadixClient().RadixV1().RadixApplications(utils.GetAppNamespace(appName)).Get(context.TODO(), appName, metav1.GetOptions{})
+}
+
+func isWhitelisted(rr *v1.RadixRegistration) bool {
+	whitelist := getWhitelist()
+	for _, item := range whitelist {
+		if rr.Name == item {
+			return true
+		}
+	}
+	return false
+}
+
+func rrIsInactive(rds []v1.RadixDeployment, rjs []v1.RadixJob, inactivityLimit time.Duration, action string) (bool, error) {
+	if len(rds) == 0 {
+		log.Debugf("no RadixDeployments found, assuming RadixRegistration is inactive")
+		return true, nil
+	}
+	latestRadixDeployment := SortDeploymentsByActiveFromTimestampAsc(rds)[len(rds)-1]
+	latestRadixDeploymentTimestamp := latestRadixDeployment.Status.ActiveFrom
+	latestRadixJobTimestamp := metav1.Time{Time: time.Unix(0, 0)}
+	latestRadixJob := getLatestRadixJob(rjs)
+	if latestRadixJob != nil {
+		latestRadixJobTimestamp = *latestRadixJob.Status.Created
+	}
+
+	log.Debugf("most recent radixDeployment is %s, active from %s, %d hours ago", latestRadixDeployment.Name, latestRadixDeploymentTimestamp.Format(time.RFC822), int(time.Now().Sub(latestRadixDeploymentTimestamp.Time).Hours()))
+	log.Debugf("most recent radixJob was %s, created %s, %d hours ago", latestRadixJob.Name, latestRadixJobTimestamp.Format(time.RFC822), int(time.Now().Sub(latestRadixJobTimestamp.Time).Hours()))
+	latestUserMutationTimestamp, err := getLastUserMutationTimestamp(latestRadixDeployment)
+	if err != nil {
+		return false, err
+	}
+	log.Debugf("most recent manual user activity was %s, %d hours ago", latestUserMutationTimestamp.Format(time.RFC822), int(time.Now().Sub(latestUserMutationTimestamp.Time).Hours()))
+	lastActivity := getMostRecentTimestamp(&latestRadixJobTimestamp, latestUserMutationTimestamp, &latestRadixDeploymentTimestamp)
+	if tooLongInactivity(lastActivity, inactivityLimit) {
+		log.Infof("%s: last activity was %d hours ago, which is more than %d hours ago, marking for %s", latestRadixDeployment.Spec.AppName, int(time.Now().Sub(lastActivity.Time).Hours()), int(inactivityLimit.Hours()), action)
+		return true, nil
```

Comment:
Maybe collect errors and `continue` - not to break the entire loop on each error